content/client-lib-development-guide/features.textile

```diff
@@ -339,6 +340,14 @@ h3(#rest-encryption). Encryption
 h3(#rest-compatibility). Forwards compatibility
 * @(RSF1)@ The library must apply the "robustness principle":https://en.wikipedia.org/wiki/Robustness_principle in its processing of requests and responses with the Ably system. In particular, deserialization of Messages and related types, and associated enums, must be tolerant to unrecognised attributes or enum values. Such unrecognised values must be ignored.
 
+h3(#batch-operations). Batch Operations
```

Comment:
We should maybe add something to say that batch operations "best effort", and not transactional (all or nothing) in nature.

Like If a user was to publish a batch of 2 messages. If the first publish failed, an attempt would still be made to publish the second message.  The reason to add this info would be that in some contexts, batch operations (like database writes) the batch can be transactional where either everything is written or nothing is written.

Reply:
Would this be a consideration from an SDK perspective? I would imagine that the handling for the returned value is the same as the transactional nature is only relevant to the server-side implementation. If there is a difference in client-side behavior that you can foresee I will put a spec point in.

Reply:
Thinking about it now, I'm not sure how a publish operation could be transactional as the SDK only knows if a publish has failed when it tries to publish, by which time the message has gone and can't be rolled back.  I guess it's inferred that all the batch operations for publish are "best effort" as they are sending data.